### PR TITLE
Use quoted elements for reporting test/valid cols.

### PR DESCRIPTION
**Pull request by Pscheidl**
Originally reported by @arnocandel, the following output is confusing:

```
09-11 22:03:26.496 192.168.86.47:54321   90536    FJ-1-101  WARN water.default: Test/Validation dataset column 'L0_S22_F542' has levels not trained on: []
09-11 22:03:26.496 192.168.86.47:54321   90536    FJ-1-101  WARN water.default: Test/Validation dataset column 'L0_S22_F544' has levels not trained on: []
09-11 22:03:26.496 192.168.86.47:54321   90536    FJ-1-101  WARN water.default: Test/Validation dataset column 'L0_S22_F545' has levels not trained on: [, T16384]
09-11 22:03:26.496 192.168.86.47:54321   90536    FJ-1-101  WARN water.default: Test/Validation dataset column 'L0_S22_F547' has levels not trained on: []
09-11 22:03:26.496 192.168.86.47:54321   90536    FJ-1-101  WARN water.default: Test/Validation dataset column 'L0_S22_F549' has levels not trained on: []
09-11 22:03:26.496 192.168.86.47:54321   90536    FJ-1-101  WARN water.default: Test/Validation dataset column 'L0_S22_F550' has levels not trained on: [, T16384]
09-11 22:03:26.496 192.168.86.47:54321   90536    FJ-1-101  WARN water.default: Test/Validation dataset column 'L0_S22_F552' has levels not trained on: []
09-11 22:03:26.496 192.168.86.47:54321   90536    FJ-1-101  WARN water.default: Test/Validation dataset column 'L0_S22_F554' has levels not trained on: []
09-11 22:03:26.496 192.168.86.47:54321   90536    FJ-1-101  WARN water.default: Test/Validation dataset column 'L0_S22_F555' has levels not trained on: [, T16384]
09-11 22:03:26.496 192.168.86.47:54321   90536    FJ-1-101  WARN water.default: Test/Validation dataset column 'L0_S22_F557' has levels not trained on: []
09-11 22:03:26.496 192.168.86.47:54321   90536    FJ-1-101  WARN water.default: Test/Validation dataset column 'L0_S22_F559' has levels not trained on: []
09-11 22:03:26.496 192.168.86.47:54321   90536    FJ-1-101  WARN water.default: Test/Validation dataset column 'L0_S22_F560' has levels not trained on: [, T16384]
```

as there are empty values treated as categorical levels. Which is probably desired. To make it more readable, I made this PR.
Current output:

```
09-14 11:51:00.918 192.168.0.149:54321   12863     FJ-1-15  INFO water.default: Test/Validation dataset column 'C1' has levels not trained on: ["", "T16384"]
```

### CODE REVIEW

File: h2o-core/src/main/java/water/util/ArrayUtils.java [hunk 1]
```diff
@@ -667,6 +667,26 @@ public static String arrayToString(int[] ary) {
...
+      return "[]";
+
+    StringBuilder b = new StringBuilder();
+    b.append('[');
+    for (int i = 0; ; i++) {
+      b.append('"');
+      b.append(a[i]);
+      b.append('"');
+      if (i == iMax)
+        return b.append(']').toString();
```
Comment (honzasterba) [inline h2o-core/src/main/java/water/util/ArrayUtils.java:685]: It might be my personal preference, but this return inside-if-inside-for is not very readable to me.
also the missing end condition in the for loop
I would suggest replacing this if with `if (i != iMax) b.append(", ");` and moving the return to the end of method

File: h2o-core/src/test/java/water/util/ArrayUtilsTest.java [hunk 2]
```diff
@@ -267,4 +267,11 @@ public void testByteArrayEmptySelect() {
     assertArrayEquals("Selected array elements mismatch.",
                       expectedSelectedElements, ArrayUtils.select(arr, idxs));
   }
+  
+  @Test
+  public void testToStringQuotedElements(){
```
Comment (honzasterba) [inline h2o-core/src/test/java/water/util/ArrayUtilsTest.java:272]: would also test corner-cases, like empty input

File: h2o-core/src/main/java/water/util/ArrayUtils.java (hunk 1 above)
Comment (satai) [reply h2o-core/src/main/java/water/util/ArrayUtils.java:685]: Since this is not a performance critical part of code, my two cents to:

```public static String toStringQuotedElements(Object[] a) {
    if (a == null)
      return "null";
    if (0 == a.length)
      return "[]";
    return "[\"" + String.join("\",\"", new String [] { "foo", "bar" }) + "\"]";
  }```

**Comment by michalkurka** [comment]
thank you @Pscheidl - please merge to master instead

**Review by michalkurka — APPROVED** [review]
